packages/store/addon/-private/system/ds-model-store.ts

```diff
@@ -143,7 +144,7 @@ class Store extends CoreStore {
   @param {String} modelName
   @return {Model}
     */
-  modelFor(modelName) {
+  modelFor(modelName: string): ShimModelClass | Model {
```

Comment:
I’ll figure out how to make sure TS understands this is a type that extends Model

Reply:
http://www.typescriptlang.org/play/#code/MYGwhgzhAECiAeYC2AHEBTaBvAUNaEALmIQJbDQAW6IIA9gFwGEBOpAdgObQC80A5AEYADKOH8A3HmgB3OixAATJkTZdeAgMz8cAXxw4AZgFd2wMnXbQW6QgmRp0ACgCUTQgE8U6OobiJUDGxpG0JjFit7QPQpfRxgSyJoMA1QqMdXKTAAOmpaOikgA

If `Model` were a class and was typed then we would use `typeof`, however because it comes from JS it is an `any` and `typeof any` is unhappy. This is something we will need to fix once we can type `model`.

I spiked typing `Model`, however, until we can extend using class syntax this problem will persist.